### PR TITLE
split the long requirement on decoding chunked trailer fields into …

### PR DESCRIPTION
**Pull request by royfielding**
…three separate requirements specific to each decision

For #966

### CODE REVIEW

File: draft-ietf-httpbis-messaging-latest.xml [hunk 1]
```diff
@@ -1380,13 +1380,15 @@ Transfer-Encoding: gzip, chunked
...
-   any received trailer fields, store/forward them separately from the header
-   fields, or selectively merge into the header section only those trailer
-   fields corresponding to header field definitions that are understood by
-   the recipient to explicitly permit and define how their corresponding
-   trailer field value can be safely merged.
+   A recipient that removes the chunked encoding from a message &MAY;
+   selectively retain or discard the received trailer fields. A recipient
+   that retains a received trailer field &MUST; either store/forward the
+   trailer field separately from the received header fields (e.g., within a
+   separate table or trailer section) or merge the received trailer field
```
Comment (reschke) [inline draft-ietf-httpbis-messaging-latest.xml:1387]: separate "table"?

**Review by reschke — CHANGES_REQUESTED** [review]

File: draft-ietf-httpbis-messaging-latest.xml (hunk 1 above)
Comment (royfielding) [reply draft-ietf-httpbis-messaging-latest.xml:1387]: yes, that's one example

Comment (royfielding) [reply draft-ietf-httpbis-messaging-latest.xml:1387]: they can be stored as a separate lookup table or forwarded as a separate trailer section

File: draft-ietf-httpbis-messaging-latest.xml [hunk 2]
```diff
@@ -1380,13 +1380,15 @@ Transfer-Encoding: gzip, chunked
...
+   into the header section. A recipient &MUST-NOT; merge a received trailer
+   field into the header section unless its corresponding header field
+   definition explicitly permits and instructs how the trailer field value
+   can be safely merged.
```
Comment (royfielding) [inline draft-ietf-httpbis-messaging-latest.xml:1391]:
```suggestion
   A recipient that removes the chunked encoding from a message &MAY;
   selectively retain or discard the received trailer fields. A recipient
   that retains a received trailer field &MUST; either store/forward the
   trailer field separately from the received header fields or merge the
   received trailer field into the header section.
   A recipient &MUST-NOT; merge a received trailer field into the
   header section unless its corresponding header field definition
   explicitly permits and instructs how the trailer field value can be
   safely merged.
```

File: draft-ietf-httpbis-messaging-latest.xml (hunk 1 above)
Comment (royfielding) [reply draft-ietf-httpbis-messaging-latest.xml]: easier to just remove the example

**Review by reschke — APPROVED** [review]

**Review by mnot — APPROVED** [review]

**Pull request merged**